EIPS/eip-3332.md

```diff
@@ -0,0 +1,95 @@
+---
+eip: 3332
+title: MEDGASPRICE Opcode
+author: Justice Hudson (@jchancehud)
+discussions-to: https://ethereum-magicians.org/t/medgasprice-opcode-eip/5480
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-05
+---
+
+## Simple Summary
+
+Add `MEDGASPRICE (0x46)` that allows smart contracts to access information about current gas prices.
+
+## Abstract
+
+There is an existing opcode `GASPRICE (0x3a)` that allows contracts to access the gas price of the current transaction. This value can be used to impose a maximum gas price on contract calls by setting a constant value, but cannot be used as a long term solution given the fluctuation of gas prices over time. The proposed opcode `MEDGASPRICE (0x46)` would return the median gas price for the parent block. This would allow the implementation of upper bounds on transaction gas prices as a function of the last block median gas price.
```

Comment:
```suggestion
Adds `MEDGASPRICE (0x46) opcode that returns the median gas price for the parent block.
```
The abstract should be a human readable version of the specification.  In this case, the specification is very simple so the abstract ends up being quite simple/short.  I also suggest altering the simple summary to be *less* specific (even more generalized).  What is in the abstract currently is mostly motivation, which should be in the Motivation section.